### PR TITLE
User creation form 

### PR DESCRIPTION
**Pull request by Gamid-FrontEnd**

### CODE REVIEW

File: js/chat_app/src/userCreationForm/user_create_form.css [hunk 1]
```diff
@@ -0,0 +1,77 @@
+.form {
```
Comment (zoolamyke) [inline js/chat_app/src/userCreationForm/user_create_form.css:1]: File should be renamed

File: js/chat_app/src/userCreationForm/UserCreationForm.jsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        "Content-type": "application/json",
+        "Auth-Token": adminToken,
+      },
+    });
+    const json = await response.json();
+    setUsers(json);
+  }
+
+  return (
+    <div className={"form" + styleForm}>
```
Comment (zoolamyke) [inline js/chat_app/src/userCreationForm/UserCreationForm.jsx:65]: Don't concatenate strings with +.
Use interpolation.
```suggestion
    <div className={`form${styleForm}`}>
```

File: js/chat_app/src/userCreationForm/UserCreationForm.jsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        "Content-type": "application/json",
+        "Auth-Token": adminToken,
+      },
+      body: JSON.stringify({
+        adminId: adminId,
+        username: username,
+        password: password,
+      }),
+    });
+    getUsers(adminToken);
```
Comment (zoolamyke) [inline js/chat_app/src/userCreationForm/UserCreationForm.jsx:49]: This is a serious problem - things have to be done in sequence, not in parallel.
```suggestion
    fetch(...).then(() => {
        getUsers(adminToken);
    })

```

File: js/chat_app/src/userCreationForm/UserCreationForm.jsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+          placeholder="Password"
+          type="password"
+          value={passwordInput}
+          onChange={(e) => setPasswordInput(e.target.value)}
+        />
+        <button>SEND</button>
+      </form>
+      <div>
+        <ul className="userListUl">
+          {users
```
Comment (zoolamyke) [inline js/chat_app/src/userCreationForm/UserCreationForm.jsx:96]: This is more concise. Read more about `??` [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)
```suggestion
          users?.map(...) ?? 'Loading...'
```

File: js/chat_app/src/userCreationForm/UserCreationForm.jsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            ? users.map((user) => {
+                if (
+                  user.username === "admin" &&
+                  token.user.id === "fhs8dhf9s8dhf9sd8hf9sd8hf"
+                ) {
+                  return (
+                    <li key={user.id} className="userListLi">
+                      <Avatar sx={{ bgcolor: green[500] }} variant="square">
+                        {user.username.charAt(0)}
```
Comment (zoolamyke) [inline js/chat_app/src/userCreationForm/UserCreationForm.jsx:105]: Here, and in other places.

```suggestion
                        {user.username.charAt(0).toUpperCase()}
```

File: js/chat_app/src/userCreationForm/UserCreationForm.jsx (hunk 2 above)
Comment (zoolamyke) [inline js/chat_app/src/userCreationForm/UserCreationForm.jsx:65]: You should be using `Dialog` component from Material UI.

File: js/chat_app/src/index.js [hunk 6]
```diff
@@ -1,17 +1,14 @@
...
+const root = ReactDOM.createRoot(document.getElementById("root"));
 root.render(
-  <React.StrictMode>
+  <>
     <App />
-  </React.StrictMode>
+  </>
 );
 
-// If you want to start measuring performance in your app, pass a function
```
Comment (zoolamyke) [inline js/chat_app/src/index.js:14]: These comments shouldn't have been removed.

File: js/chat_app/src/components/userCreationForm/UserCreationForm.jsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+          onChange={(e) => setPasswordInput(e.target.value)}
+        />
+        <button>SEND</button>
+      </form>
+      <div>
+        <ul className="userListUl">
+          {users?.map((user) => {
+            if (
+              user.username === "admin" &&
+              auth.user.id === "fhs8dhf9s8dhf9sd8hf9sd8hf"
```
Comment (zoolamyke) [inline js/chat_app/src/components/userCreationForm/UserCreationForm.jsx:80]: remove hardcode

File: js/chat_app/src/components/userCreationForm/UserCreationForm.jsx [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        "Content-type": "application/json",
+        "Auth-Token": adminToken,
+      },
+    });
+    const json = await response.json();
+    setUsers(json);
+  }
+
+  return (
+    <div className={`form ${styleForm}`}>
```
Comment (zoolamyke) [inline js/chat_app/src/components/userCreationForm/UserCreationForm.jsx:46]:
```suggestion
    import classnames from 'classnames'

    <div className={classnames('form', 'styleForm', { 'is-visible': isVisible }})>
```

File: js/chat_app/src/components/DashBoard.js [hunk 9]
```diff
@@ -53,9 +54,10 @@ export const DashBoard = () => {
             />
           </Toolbar>
         </AppBar>
+        {<UserCreationButton />}
```
Comment (miromyke) [inline js/chat_app/src/components/DashBoard.js]:
```suggestion
        <UserCreationButton />
```

File: js/chat_app/src/components/userCreationForm/UserCreationButton.jsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  const { auth } = useContext(UserContext);
+  const [open, setOpen] = useState(false);
+
+  return (
+    <div className="userCreate">
+      <Button
+        variant="contained"
+        onClick={() => setOpen(true)}
+        className="userCreateButton"
+        disabled={auth.isAdmin ? false : true}
```
Comment (miromyke) [inline js/chat_app/src/components/userCreationForm/UserCreationButton.jsx]:
```suggestion
        disabled={auth.isAdmin}
```

File: js/chat_app/src/ChatList.js [hunk 11]
```diff
@@ -1,52 +0,0 @@
-import React, { useContext } from "react";
```
Comment (miromyke) [inline js/chat_app/src/ChatList.js:1]: I assume it's a duplicated file?

File: js/chat_app/package.json [hunk 12]
```diff
@@ -6,10 +6,10 @@
     "@emotion/react": "^11.10.4",
     "@emotion/styled": "^11.10.4",
     "@icons/material": "^0.4.1",
-    "@material-ui/styles": "^4.11.5",
     "@mui/icons-material": "^5.10.6",
     "@mui/lab": "^5.0.0-alpha.101",
-    "@mui/material": "^5.10.7",
+    "@mui/material": "^5.10.9",
```
Comment (miromyke) [inline js/chat_app/package.json:7]: Dependencies need to be reconsidered.

File: js/chat_app/src/components/userCreationForm/userCreationForm.css [hunk 13]
```diff
@@ -0,0 +1,34 @@
+
+
+.userCreate {
+    width: 135px;
```
Comment (miromyke) [inline js/chat_app/src/components/userCreationForm/userCreationForm.css]: Let's get away from absolute width values.

**Pull request merged**